torch/utils/data/datapipes/iter/combinatorics.py

```diff
@@ -86,6 +87,7 @@ def __init__(self,
         else:
             self.datapipe = datapipe.unbatch(unbatch_level=unbatch_level)
         self.buffer_size = buffer_size
+        self._shuffle_enabled = True
```

Comment:
As discussed in our latest sync, please add an option to set this when creating the `Shuffler`. That way we can include the `Shuffler` in "the right spot" in the pipeline without actually shuffling until explicitly requested.